CMakeLists.txt

```diff
@@ -40,9 +40,20 @@ include(${TORCH_XPU_OPS_ROOT}/cmake/ONEMKL.cmake)
 include(${TORCH_XPU_OPS_ROOT}/cmake/BuildFlags.cmake)
 
 option(USE_XCCL "Build with XCCL support" OFF)
+set(USE_C10D_XCCL OFF CACHE BOOL "Build with XCCL support for C10D")
 
-if(NOT WIN32 AND USE_XCCL)
+if(USE_XCCL)
   include(${TORCH_XPU_OPS_ROOT}/cmake/XCCL.cmake)
+  if(NOT PYTORCH_FOUND_XCCL)
+    caffe2_update_option(USE_XCCL OFF)
+    # -- [ Re-generate the macros file for https://github.com/pytorch/pytorch/pull/147161
+    configure_file(
+      ${TORCH_XPU_OPS_ROOT}/../../caffe2/core/macros.h.in
+      ${CMAKE_BINARY_DIR}/caffe2/core/macros.h)
+  else()
+    # Propagate the option to PyTorch
+    caffe2_update_option(USE_C10D_XCCL ON)
+  endif()
 endif()
```

Comment:
@guangyey can you also propagate the option to PyTorch when `USE_XCCL=OFF` also? I believe this will fix my failing unit test: 
https://github.com/pytorch/pytorch/actions/runs/13425292655/job/37507957344?pr=147161#step:14:9872
when your PR is merged.

Reply:
`USE_C10D_XCCL` is defined as `OFF` by default in cmake. So if `USE_XCCL=OFF`, `USE_C10_XCCL` will be `OFF`. It is unnecessary to update it, right?

Reply:
I think if we don't update it we end up with `USE_XCCL=` in `torch.__config__.show()` because the user will (normally) not explicitly use `export USE_XCCL=0`.

Reply:
I see your point.

Reply:
updated in line 45

Reply:
I think we need also to regenerate the macros file when `USE_XCCL` is not defined like you did in the other case?

Reply:
Good catch!